sound/soc/intel/boards/sof_pcm512x.c

```diff
@@ -136,6 +139,9 @@ static int sof_card_late_probe(struct snd_soc_card *card)
 	if (list_empty(&ctx->hdmi_pcm_list))
 		return -EINVAL;
 
+	if (!ctx->idisp_codec)
+		return 0;
+
```

Comment:
also unrelated: it isn't very obvious either why just taking the first member in the list is good. That means, that `hda_dsp_hdmi_build_controls()` can be called with any codec DAI component? Looks a bit confusing to me.

Reply:
@lyakh This is based on the assumption that all HDMI PCMs are handled by the same codec, so we look up the driver via the codec_dai->component of the first PCM. I guess a comment would not hurt.